trieste/models/interfaces.py

```diff
@@ -90,9 +90,11 @@ def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
             f"Model {self!r} does not support predicting observations, just the latent function"
         )
 
-    def log(self) -> None:
+    def log(self, dataset: Dataset) -> None:
         """
         Log model-specific information at a given optimization step.
+
+        :param dataset: The data that can be used to log additional data-based model summaries.
```

Comment:
@uri-granta added dataset argument as its more flexible, there are some metrics we would like to log that are based on the training data

Reply:
perhaps we should make it optional though?

Reply:
Yes, make this optional, and default it to None (like acquisition function builders)

Reply:
done!